### DISCUSSION TITLE
How to get same CID as web3.storage?

### DISCUSSION THREAD

**Discussion by helloivanco** [web3-storage/web3.storage#2231] [Q&A]
By using the Multiformats library, I'm able to get the same CID for files under 1MB with this function. However, for files greater than 1MB it is returning a different CID.

```
async function getCid(buffer: Buffer) {
  const bytes = new Uint8Array(buffer);
  const hash = await sha256.digest(bytes);
  const cid = CID.createV1(raw.code, hash);
  return cid.toString();
}
```

I've also tried to packToBlob with ipfs-car but get much different CIDs.

```
 const { car } = await packToBlob({
    input: [buffer],
    maxChunkSize: MAX_CHUNK_SIZE,
    maxChildrenPerNode: MAX_CHILDREN_PER_NODE,
    wrapWithDirectory: false,
  });
  const carBytes = new Uint8Array(await car.arrayBuffer());
  const hash = await sha256.digest(carBytes);
  const cid = CID.createV1(raw.code, hash);
```

Any help figuring this out would be appreciated. Btw, a simple function or library to determine the CID v1 of a file would be a useful tool to prevent filename collisions and sets one up for content addressing.

**Comment by sudeepdino008** [comment]
you have to use the same merkle dag creation parameters for this. I have a related article on this - https://moskud.xyz/cids

The steps are:
1. Generate car files with fixed parameters (like chunk-size or tree-structure)
2. upload/add those car files to ipfs. 

Wherever you generate the car file, if you can use the same parameters, you are set. 

You can try to unpack the code here - https://github.com/covalenthq/ipfs-pinner/blob/main/binary/main.g,o

Let me know if this helps.

**Comment by helloivanco** [comment]
Thank you for following up!

These parameters worked:
`  wrapWithDirectory: false,
    maxChunkSize: 1048576,
    maxChildrenPerNode: 1024,
`

I ended up making this npm package which takes a file buffer and returns the proper CID. https://www.npmjs.com/package/ipfs-web3storage-cid
